src/umpire/Allocator.hpp

```diff
@@ -68,7 +68,7 @@ class Allocator : private strategy::mixins::Inspector, strategy::mixins::Allocat
   /*!
    * \brief Free the memory at ptr.
    *
-   * This method will throw an umpire::Exception if ptr was not allocated
+   * This method will throw an umpire::runtime_error if ptr was not allocated
```

Comment:
Do we still even have `umpire::Exception`s?  It seems like we may want to remove its implementation?